stubs/invoke/invoke/tasks.pyi

```diff
@@ -1,57 +1,75 @@
 from _typeshed import Self
-from typing import Any
+from typing import Any, Callable, Iterable, TypeVar
 
 from .config import Config
 from .context import Context
+from .parser import Argument
 
 NO_DEFAULT: object
 
 class Task:
-    body: Any
-    __doc__: str
+    body: Callable[..., Any]
+    __doc__: str | None
     __name__: str
-    __module__: Any
-    aliases: Any
+    __module__: str
+    aliases: tuple[str, ...]
     is_default: bool
-    positional: Any
-    optional: Any
-    iterable: Any
-    incrementable: Any
-    auto_shortflags: Any
-    help: Any
-    pre: Any
-    post: Any
+    positional: Iterable[str]
+    optional: Iterable[str]
+    iterable: Iterable[str]
+    incrementable: Iterable[str]
+    auto_shortflags: bool
+    help: dict[str, str]
+    pre: list[Task]
+    post: list[Task]
     times_called: int
-    autoprint: Any
+    autoprint: bool
     def __init__(
         self,
-        body,
-        name=...,
-        aliases=...,
-        positional=...,
-        optional=...,
+        body: Callable[..., Any],
+        name: str | None = ...,
+        aliases: tuple[str, ...] = ...,
+        positional: Iterable[str] | None = ...,
+        optional: Iterable[str] = ...,
         default: bool = ...,
         auto_shortflags: bool = ...,
-        help=...,
-        pre=...,
-        post=...,
+        help: dict[str, str] | None = ...,
+        pre: list[Task] | None = ...,
+        post: list[Task] | None = ...,
         autoprint: bool = ...,
-        iterable=...,
-        incrementable=...,
+        iterable: Iterable[str] | None = ...,
+        incrementable: Iterable[str] | None = ...,
     ) -> None: ...
     @property
     def name(self): ...
-    def __eq__(self, other): ...
-    def __hash__(self): ...
+    def __eq__(self, other: object) -> bool: ...
```

Comment:
It seems like this will actually raise `AttributeError` if you try to compare a `Task` with something that doesn't have `name` or `body` attributes. https://github.com/pyinvoke/invoke/blob/45dc9d03639dac5b6d1445831bf270e686ef88b4/invoke/tasks.py#L100

```suggestion
    def __eq__(self, other: Task) -> bool: ...  # type: ignore[override]
```